source/fe/mapping_isoparametric.cc

```diff
@@ -0,0 +1,2235 @@
+       update_jacobian_pushed_forward_2nd_derivatives |
+       update_jacobian_3rd_derivatives |
+       update_jacobian_pushed_forward_3rd_derivatives))
+    shape_derivatives.resize(n_shape_functions * n_q_points);
+
+  if (this->update_each &
+      (update_jacobian_grads | update_jacobian_pushed_forward_grads))
+    shape_second_derivatives.resize(n_shape_functions * n_q_points);
+
+  if (this->update_each & (update_jacobian_2nd_derivatives |
+                           update_jacobian_pushed_forward_2nd_derivatives))
+    shape_third_derivatives.resize(n_shape_functions * n_q_points);
+
+  if (this->update_each & (update_jacobian_3rd_derivatives |
+                           update_jacobian_pushed_forward_3rd_derivatives))
+    shape_fourth_derivatives.resize(n_shape_functions * n_q_points);
+
+  // now also fill the various fields with their correct values
+  compute_shape_function_values(q.get_points());
+
+  // copy (projected) quadrature weights
+  quadrature_weights = q.get_weights();
+}
+
+
+
+template <int dim, int spacedim>
+void
+MappingIsoparametric<dim, spacedim>::InternalData::initialize_face(
+  const UpdateFlags      update_flags,
+  const Quadrature<dim> &q,
+  const unsigned int     n_original_q_points)
+{
+  initialize(update_flags, q, n_original_q_points);
+
+  if (this->update_each &
+      (update_boundary_forms | update_normal_vectors | update_jacobians |
+       update_JxW_values | update_inverse_jacobians))
+    {
+      aux.resize(dim - 1,
+                 std::vector<Tensor<1, spacedim>>(n_original_q_points));
+
+      // Compute tangentials to the unit cell.
+      if (this->fe.reference_cell_type() == ReferenceCell::get_hypercube(dim))
+        {
+          for (const unsigned int i : GeometryInfo<dim>::face_indices())
+            {
+              unit_tangentials[i].resize(n_original_q_points);
+              std::fill(unit_tangentials[i].begin(),
+                        unit_tangentials[i].end(),
+                        GeometryInfo<dim>::unit_tangential_vectors[i][0]);
+              if (dim > 2)
+                {
+                  unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                    .resize(n_original_q_points);
+                  std::fill(
+                    unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                      .begin(),
+                    unit_tangentials[GeometryInfo<dim>::faces_per_cell + i]
+                      .end(),
+                    GeometryInfo<dim>::unit_tangential_vectors[i][1]);
+                }
+            }
+        }
+      else if (this->fe.reference_cell_type() == ReferenceCell::Type::Tri)
+        {
+          Tensor<1, dim> t1;
+          t1[0] = 1;
+          t1[1] = 0;
+          for (unsigned int i = 0; i < n_original_q_points; i++)
+            unit_tangentials[0].emplace_back(t1);
+          t1[0] = -0.70710678118;
+          t1[1] = 0.70710678118;
```

Comment:
I have not checked all details but this caught my eyes because the last digit is not correctly rounded, assuming that you want to do this:
```suggestion
          t1[0] = -std::sqrt(0.5);
          t1[1] = std::sqrt(0.5);
```